libgalois/include/galois/graphs/PropertyFileGraph.h

```diff
@@ -202,14 +203,68 @@ class GALOIS_EXPORT PropertyFileGraph {
     return rdg_.edge_table()->column(i);
   }
 
-  std::shared_ptr<arrow::ChunkedArray> NodeProperty(std::string name) const {
+  /**
+   * Get a node property by name.
+   *
+   * @param name The name of the property to get.
+   * @return The property data or NULL if the property is not found.
+   */
+  std::shared_ptr<arrow::ChunkedArray> NodeProperty(
+      const std::string& name) const {
     return rdg_.node_table()->GetColumnByName(name);
   }
 
-  std::shared_ptr<arrow::ChunkedArray> EdgeProperty(std::string name) const {
+  std::shared_ptr<arrow::ChunkedArray> EdgeProperty(
+      const std::string& name) const {
     return rdg_.edge_table()->GetColumnByName(name);
   }
 
+  /**
+   * Get a node property by name specifying it's type.
+   * @tparam T The type of the property.
+   * @param name The name of the property.
+   * @return The property array or an error if the property does not exist or has a different type.
+   */
+  template <typename T>
+  Result<std::shared_ptr<typename arrow::CTypeTraits<T>::ArrayType>>
+  NodePropertyTyped(const std::string& name) {
+    auto chunked_array = NodeProperty(name);
+    if (!chunked_array) {
+      return ErrorCode::PropertyNotFound;
+    }
+
+    auto array =
+        std::dynamic_pointer_cast<typename arrow::CTypeTraits<T>::ArrayType>(
+            chunked_array->chunk(0));
+    if (!array) {
+      return ErrorCode::TypeError;
+    }
+    return array;
+  }
+
+  /**
+    * Get an edge property by name specifying it's type.
+    * @tparam T The type of the property.
+    * @param name The name of the property.
+    * @return The property array or an error if the property does not exist or has a different type.
+    */
+  template <typename T>
+  Result<std::shared_ptr<typename arrow::CTypeTraits<T>::ArrayType>>
+  EdgePropertyTyped(const std::string& name) {
+    auto chunked_array = EdgeProperty(name);
+    if (!chunked_array) {
+      return ErrorCode::PropertyNotFound;
+    }
+
+    auto array =
+        std::dynamic_pointer_cast<typename arrow::CTypeTraits<T>::ArrayType>(
+            chunked_array->chunk(0));
+    if (!array) {
+      return ErrorCode::TypeError;
+    }
+    return array;
+  }
```

Comment:
@danielmawhirter Could I get you to look at these new functions? There was some debate and I wanted to make sure this was OK with people.

Reply:
Looks good to me! Only note is I tend to LOG something at an error's origin, but not sure if that's standardized through the codebase.

Reply:
The logs are not actually useful because they will not be visible in all cases (for instance they don't reliably show up in Jupyter notebooks when called from the Python API). Also the logged message will be spurious if the error is handled by the caller and not propagated further. The idea going forward is to attach message and location information to the error return value itself. I think we should probably push forward with that soon and start standardizing.